openjdk/src/main/java/org/conscrypt/NativeCryptoJni.java

```diff
@@ -16,126 +16,97 @@
-        //
-        // TODO(davidben): Pick one name for the library and stick with it.
-        String platformSpecific = LIB_NAME + "-" + os + '-' + arch;
-        NativeLibraryLoader.loadFirstAvailable(
-                NativeCrypto.class.getClassLoader(), platformSpecific, LIB_NAME);
+    /**
+     * Attempts to load the shared JNI library. First try loading the platform-specific library
+     * name (e.g. conscrypt_openjdk_jni-linux-x86_64). If not found, try the static library name.
+     *
+     * The static library name is needed in order to support Java 8 static linking
+     * (http://openjdk.java.net/jeps/178), where the library name is used to invoke a
+     * library-specific load method (i.e. {@code JNI_OnLoad_conscrypt}).
+     *
+     * @throws UnsatisfiedLinkError if the library failed to load.
+     */
+    static void init() throws UnsatisfiedLinkError {
+        List<LoadResult> results = new ArrayList<>();
+        if (!NativeLibraryLoader.loadFirstAvailable(
+                    classLoader(), results, platformLibName(), STATIC_LIB_NAME)) {
+            logResults(results);
+            throwBestError(results);
+        }
     }
 
     private NativeCryptoJni() {}
 
-    /**
-     * Normalizes the os.name value into the value used by the Maven os plugin
-     * (https://github.com/trustin/os-maven-plugin). This plugin is used to generate
-     * platform-specific
-     * classifiers for artifacts.
-     */
-    private static String normalizeOs(String value) {
-        value = normalize(value);
-        if (value.startsWith("aix")) {
-            return "aix";
-        }
-        if (value.startsWith("hpux")) {
-            return "hpux";
-        }
-        if (value.startsWith("os400")) {
-            // Avoid the names such as os4000
-            if (value.length() <= 5 || !Character.isDigit(value.charAt(5))) {
-                return "os400";
-            }
-        }
-        if (value.startsWith(LINUX)) {
-            return LINUX;
-        }
-        if (value.startsWith("macosx") || value.startsWith("osx")) {
-            return "osx";
-        }
-        if (value.startsWith("freebsd")) {
-            return "freebsd";
-        }
-        if (value.startsWith("openbsd")) {
-            return "openbsd";
-        }
-        if (value.startsWith("netbsd")) {
-            return "netbsd";
-        }
-        if (value.startsWith("solaris") || value.startsWith("sunos")) {
-            return "sunos";
-        }
-        if (value.startsWith("windows")) {
-            return "windows";
+    private static void logResults(List<LoadResult> results) {
+        for (LoadResult result : results) {
+            result.log();
         }
+    }
+
+    private static void throwBestError(List<LoadResult> results) {
+        Collections.sort(results, ErrorComparator.INSTANCE);
+        // Guaranteed to have at least one UnsatisfiedLinkError - use the best one.
+        throw (UnsatisfiedLinkError) results.get(0).error;
```

Comment:
Yeah, adding them to Throwable.addSuppressed() is a good idea, though that's a 1.7 addition so we'll have to use reflection to do it.